### ISSUE TITLE
Our CodeSnips are ugly

### ISSUE THREAD

**Issue by fabianvf** [konveyor/analyzer-lsp#241]
The `gopkg.in/yaml.v2` library is not turning our codesnips into `|` style multiline strings, it's just leaving them as string literals with `\n` in there. Ideally we'd output prettier multiline strings because in the current state it's not really human-readable at all.

**Comment by fabianvf** [comment]
The issue is as far as I can tell the library doesn't currently have any way to enforce that, if there are certain (undocumented AFAIK) characters in the string it won't use the multi-line syntax

**Comment by MdSahil-oss** [comment]
@fabianvf @pranavgaikwad I would like to work on this issue.

**Comment by pranavgaikwad** [comment]
@MdSahil-oss In the `rule-example.yaml` file, `lang-ref-001` rule is a good reproducer for this one https://github.com/konveyor/analyzer-lsp/blob/7ebd2f5efb3fbfbb5e102d80e8481ee823fdb3b2/demo-output.yaml#L210

**Comment by RipulHandoo** [comment]
Hey @fabianvf @pranavgaikwad , I have manually modified the [demo-output.yaml](https://github.com/RipulHandoo/analyzer-lsp/blob/main/demo-output.yamlrl)
And pulled a    [PR](https://github.com/konveyor/analyzer-lsp/pull/285) but getting error .
How to overcome that, ```yaml``` file is valid I double-checked that

**Comment by RipulHandoo** [comment]
> Hey @fabianvf @pranavgaikwad , I have manually modified the [demo-output.yaml](https://github.com/RipulHandoo/analyzer-lsp/blob/main/demo-output.yamlrl) And pulled a [PR](https://github.com/konveyor/analyzer-lsp/pull/285) but getting error . How to overcome that, `yaml` file is valid I double-checked that

In this
 ```
codeSnip: |
          1  package com.example.apps;
          2  
          3  import io.fabric8.kubernetes.api.model.apiextensions.v1beta1.CustomResourceDefinition;
          4  
          5  public class App 
          6  {
          7  
          8      public static void main( String[] args )
          9      {
          10          CustomResourceDefinition crd = new CustomResourceDefinition();
          11          System.out.println( crd );
          12  
          13          GenericClass<String> element = new GenericClass<String>(\"Hello world!\");
          14          element.get();
          15      }
          16  }
```
Do the line number are also have to remove line number and only code lines needs to be in the ```CodeSnips:```

**Comment by Obasoro** [comment]
@pranavgaikwad is this ticket closed or open elsewhere?

**Comment by shawn-hurley** [comment]
I believe that the issue is that there can be no white space at the start of the new line. For instance:

```1.\t test = test.Foo()\n2.\t for....``` 

This will display properly, but today we have: 

https://github.com/konveyor/analyzer-lsp/blob/main/engine/engine.go#L785-L792 this bit of code, that I think is the problem with regards to padding size. 

I don't know why or when the padding size was introduced, and if we do change the code snips we must:
1. Validate that there is no impact on kantra static reports or on the tackle2-ui before we make the change.
2. If there is impact then we have to hold the PR until we can file and properly allign the work so that there is little disruption.
